common/frame.go

```diff
@@ -770,6 +770,47 @@ func (f *Frame) uncheck(selector string, opts *FrameUncheckOptions) error {
 	return err
 }
 
+// IsChecked returns true if the first element that matches the selector
+// is checked. Otherwise, returns false.
+func (f *Frame) IsChecked(selector string, opts goja.Value) bool {
+	f.log.Debugf("Frame:IsChecked", "fid:%s furl:%q sel:%q", f.ID(), f.URL(), selector)
+
+	popts := NewFrameIsCheckedOptions(f.defaultTimeout())
+	if err := popts.Parse(f.ctx, opts); err != nil {
+		k6ext.Panic(f.ctx, "%w", err)
+	}
+	checked, err := f.isChecked(selector, popts)
+	if err != nil {
+		k6ext.Panic(f.ctx, "%w", err)
+	}
+
+	return checked
+}
+
+func (f *Frame) isChecked(selector string, opts *FrameIsCheckedOptions) (bool, error) {
+	isChecked := func(apiCtx context.Context, handle *ElementHandle) (interface{}, error) {
+		v, err := handle.isChecked(apiCtx, 0) // Zero timeout when checking state
+		if errors.Is(err, ErrTimedOut) {      // We don't care about timeout errors here!
+			return v, nil
```

Comment:
Can you add a test for this scenario as well?

Reply:
Sure, a `Frame` test would make more sense, which would be out of scope for this PR, so I guess it can wait.

Reply:
👍  I updated the main functional tests issue: https://github.com/grafana/xk6-browser/issues/241.